packages/compass-preferences-model/src/react.ts

```diff
@@ -44,11 +47,64 @@ export function usePreference<K extends keyof AllPreferences>(
   return value;
 }
 
+/**
+ * A version of usePreference hook that allows to get multiple preferences in a
+ * single object for brevity
+ *
+ * @example
+ * const {
+ *   enableShell,
+ *   readOnly,
+ * } = usePreferences(['enableShell', 'readOnly']);
+ *
+ * @param keys list of preferences keys to return as an object
+ */
+export function usePreferences<K extends (keyof AllPreferences)[]>(
```

Comment:
Seems like we're having a bunch of places where the single preference hook is used in groups of 2+, so I thought I'd add this one as an easier way to access multiple preferences at once

Reply:
Yeah, so, just for context, the only reason this wasn't part of the original implementation was that this was pretty hard to type correctly – obviously not an issue if all the options in the group have the same type, though

Reply:
Ah, interesting! I wonder, was this something that was resolved by typescript itself at some point? Seems like right now typescript is able to correctly derive the value types for each key when the keys are of mixed type

Reply:
Yeah, this seems to be working perfectly fine 🙂